src/main/java/com/champions/carsharingservice/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.champions.carsharingservice.service.impl;
+
+import com.champions.carsharingservice.dto.car.CarDto;
+import com.champions.carsharingservice.dto.car.CreateCarRequestDto;
+import com.champions.carsharingservice.exception.EntityNotFoundException;
+import com.champions.carsharingservice.mapper.CarMapper;
+import com.champions.carsharingservice.model.Car;
+import com.champions.carsharingservice.repository.CarRepository;
+import com.champions.carsharingservice.service.CarService;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CarServiceImpl implements CarService {
+    private final CarRepository carRepository;
+    private final CarMapper carMapper;
+
+    @Override
+    @Transactional
+    public CarDto save(CreateCarRequestDto request) {
+        Car car = carMapper.toEntity(request);
+        car.setType(Car.CarType.valueOf(request.type()));
+        return carMapper.toDto(carRepository.save(car));
+    }
+
+    @Override
+    @Transactional
+    public List<CarDto> getAll() {
+        return carRepository.findAll().stream()
+                .map(carMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    @Transactional
```

Comment:
Same here. Check if works without Transactional
You don't interact with DB here